app/models/metadatum.rb

```diff
@@ -352,7 +352,7 @@ def self.new_without_save(sample, key, value)
   end
 
   def validated_value
-    return instance_variable_get("#{data_type}_validated_value")
+    return self["#{data_type}_validated_value"]
```

Comment:
wait, was this change really necessary? isn't it basically doing the same thing (getting the value of the instance variable?)

Reply:
(Posting this mainly for posterity)

I did some research. It's surprisingly confusing.

`instance_variable_get("@string_validated_value")` (you need the `@`, which was missing in the code) will get the instance variable `@string_validated_value`. Unfortunately, the database value for `string_validated_value` is not stored on `@string_validated_value`, but actually on `@attributes["string_validated_value"]` (Rails magic). And if you try to access `@attributes["string_validated_value"]` directly, you'll find that it's not actually the database value,  but an internal object of type `ActiveRecord::Attribute::FromDatabase`. I think Rails intends you to call the getter method `string_validated_value` it automatically generates, via self.string_validated_value, which gives you the correct value. 

NOTE:
Instance variables are always private by default. Within a class, instance variables can only be accessed by `@foo`. If you instead type `foo`, it will first try to resolve to a local variable, and then resolve to `self.foo`, which is the getter function for `foo`. This is why  `data_type` works within the class. It is actually resolving to `self.data_type` which is the Rails auto-generated getter method for the database attribute. `@data_type` would NOT work, because that's the instance variable, and the database value is not stored there.